Lib9c/Helper/Validator.cs

```diff
@@ -70,12 +70,19 @@ public static class Validator
                 worldInformation.UpdateWorld(worldRow);
             }
 
-            if (world.IsStageCleared && stageId > world.StageClearedId + 1 ||
-                !world.IsStageCleared && stageId != world.StageBegin)
+            if (!world.IsStageCleared && stageId != world.StageBegin)
             {
                 throw new InvalidStageException(
-                    $"{addressesHex}Aborted as the stage ({worldId}/{stageId}) is not cleared; " +
-                    $"cleared stage: {world.StageClearedId}"
+                    $"{addressesHex}Aborted as the stage ({worldId}/{stageId - 1}) is not cleared; " +
```

Comment:
It is just a question. Maybe I guess it is because I have a gap in lib9c domain knowledge. Why does it subtract one from `stageId`?

When it tried to clear `3/2` (`world/stage`) but the beginning stage was `3/1`, it should show stage id `2` without any modification. Doesn't it?

Reply:
In `HackAndSlash`, Player should cleared 1/6 before play 1/7. So the `stageId` subtract one in that error message.

> When it tried to clear 3/2 (world/stage) but the beginning stage was 3/1, it should show stage id 2 without any modification. Doesn't it?

- When your avatar has been cleared 1/10,
- And if try to clear 1/20
- InvalidStageException occurred with message: Aborted as the stage (1/19) is not cleared; cleared stage: (1/10)

The "cleared stage: (1/10)" part is little bit confusing. This means "your avatar has been cleared stage (1/10), so try to clear (1/11) first"..

Reply:
> The "cleared stage: (1/10)" part is little bit confusing. This means "your avatar has been cleared stage (1/10), so try to clear (1/11) first"..

I think the adjusted version seems easy to understand than before.

But anyway it is good if @planetarium/9c-dev feels comfortable when debugging.

Reply:
I'll adjust. Thanks!